src/lava/proc/dense/process.py

```diff
@@ -144,3 +144,83 @@ def __init__(self,
                          **kwargs)
+
+
+class DelayDense(Dense):
+    def __init__(self,
+                 *,
+                 weights: np.ndarray,
+                 delays: ty.Union[np.ndarray, int],
+                 name: ty.Optional[str] = None,
+                 num_message_bits: ty.Optional[int] = 0,
+                 log_config: ty.Optional[LogConfig] = None,
+                 **kwargs) -> None:
+        """Dense, delayed connections between neurons. Realizes the following
+        abstract behavior: a_out = weights * s_in
+
+        Parameters
+        ----------
+        weights : numpy.ndarray
+            2D connection weight matrix of form (num_flat_output_neurons,
+            num_flat_input_neurons) in C-order (row major).
+
+        delays : numpy.ndarray, int
+            2D connection delay matrix of form (num_flat_output_neurons,
+            num_flat_input_neurons) in C-order (row major) or integer value if
+            the same delay should be used for all synapses.
+
+        weight_exp : int, optional
+            Shared weight exponent of base 2 used to scale magnitude of
+            weights, if needed. Mostly for fixed point implementations.
+            Unnecessary for floating point implementations.
+            Default value is 0.
+
+        num_weight_bits : int, optional
+            Shared weight width/precision used by weight. Mostly for fixed
+            point implementations. Unnecessary for floating point
+            implementations.
+            Default is for weights to use full 8 bit precision.
+
+        sign_mode : SignMode, optional
+            Shared indicator whether synapse is of type SignMode.NULL,
+            SignMode.MIXED, SignMode.EXCITATORY, or SignMode.INHIBITORY. If
+            SignMode.MIXED, the sign of the weight is
+            included in the weight bits and the fixed point weight used for
+            inference is scaled by 2.
+            Unnecessary for floating point implementations.
+
+            In the fixed point implementation, weights are scaled according to
+            the following equations:
+            w_scale = 8 - num_weight_bits + weight_exp + isMixed()
+            weights = weights * (2 ** w_scale)
+
+        num_message_bits : int, optional
+            Determines whether the Dense Process deals with the incoming
+            spikes as binary spikes (num_message_bits = 0) or as graded
+            spikes (num_message_bits > 0). Default is 0.
+        """
+
+        super().__init__(weights=weights,
+                         num_message_bits=num_message_bits,
+                         name=name,
+                         log_config=log_config,
+                         **kwargs)
+
+        self._validate_delays(weights, delays)
+        max_delay = int(np.max(delays))
+        shape = weights.shape
+
+        # Variables
+        self.delays = Var(shape=shape, init=delays)
+        self.a_buff = Var(shape=(shape[0], max_delay + 1) , init=0)
```

Comment:
It might be useful to add `max_delay` as a parameter rather than using the max delay value that we get during initialization. This works for inference, but if the user pauses the process and sets a new delay matrix with a larger delay, it might cause errors.

Reply:
Good suggestion, I added `max_delay` as a parameter.